### PR TITLE
refactor namespaces to use PathConfig

### PR DESCRIPTION
**Pull request by sethpollack**

### CODE REVIEW

File: pkg/transformers/namespace.go [hunk 1]
```diff
@@ -31,13 +60,37 @@ func NewNamespaceTransformer(ns string) Transformer {
 	if len(ns) == 0 {
 		return NewNoOpTransformer()
 	}
-	return &namespaceTransformer{namespace: ns}
+
+	sort.Sort(ByPathConfig(namespacePathConfigs))
```
Comment (monopole) [inline pkg/transformers/namespace.go]: Please add a comment explaining why order matters

File: pkg/transformers/namespace.go [hunk 2]
```diff
@@ -31,13 +60,37 @@ func NewNamespaceTransformer(ns string) Transformer {
...
+	for id := range m {
+		objMap := m[id].UnstructuredContent()
+
+		for _, path := range o.pathConfigs {
+			if !selectByGVK(id.Gvk(), path.GroupVersionKind) {
+				continue
+			}
+
+			if path.Skip {
+				break
```
Comment (monopole) [inline pkg/transformers/namespace.go]: continue?

File: pkg/transformers/namespace.go (hunk 1 above)
Comment (sethpollack) [reply pkg/transformers/namespace.go]: will do

File: pkg/transformers/namespace.go (hunk 2 above)
Comment (sethpollack) [reply pkg/transformers/namespace.go]: No if we use continue we would just go to the next path which would eventually be:
 
```
	{
		Path:               []string{"metadata", "namespace"},
		CreateIfNotPresent: true,
	},
```
And the the namespace would end up getting set.

**Comment by monopole** [comment]
Seth, it feels six one way, a half dozen the other, but #70 doesn't require the sort.  Can you adapt that for namespaces?

**Comment by sethpollack** [comment]
yup

**Comment by sethpollack** [comment]
done

**Pull request merged**